cetlvast/suites/unittest/test_any.cpp

```diff
@@ -0,0 +1,303 @@
+/// @file
+/// Unit tests for cetl/any.hpp
+///
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#include <cetl/any.hpp>
+#include <cetl/pf17/cetlpf.hpp>
+
+#include <complex>
+#include <functional>
+#include <string>
+#include <gtest/gtest.h>
+
+namespace
+{
+
+using cetl::any;
+using cetl::any_cast;
+using cetl::in_place_type_t;
+
+/// TESTS -----------------------------------------------------------------------------------------------------------
+
+TEST(test_any, cppref_example)
+{
+    using uut = any<sizeof(int)>;
+
+    uut  a{1};
+    auto ptr = any_cast<int>(&a);
+    EXPECT_TRUE(ptr);
+    EXPECT_EQ(1, *ptr);
+
+    // TODO: Add more from the example when corresponding api will be available.
+    // https://en.cppreference.com/w/cpp/utility/any
+}
+
+TEST(test_any, ctor_1_default)
+{
+    EXPECT_FALSE((any<0>{}.has_value()));
+    EXPECT_FALSE((any<0, false>{}.has_value()));
+    EXPECT_FALSE((any<0, false, true>{}.has_value()));
+    EXPECT_FALSE((any<0, true, false>{}.has_value()));
+
+    EXPECT_FALSE((any<1>{}.has_value()));
+    EXPECT_FALSE((any<1, false>{}.has_value()));
+    EXPECT_FALSE((any<1, false, true>{}.has_value()));
+    EXPECT_FALSE((any<1, true, false>{}.has_value()));
+
+    EXPECT_FALSE((any<13>{}.has_value()));
+    EXPECT_FALSE((any<13, false>{}.has_value()));
+    EXPECT_FALSE((any<13, false, true>{}.has_value()));
+    EXPECT_FALSE((any<13, true, false>{}.has_value()));
+}
+
+TEST(test_any, ctor_2_copy)
+{
+    // Primitive `int`
+    {
+        using uut = any<sizeof(int)>;
+
+        const uut srcAny{42};
```

Comment:
Applies in other places as well.

```suggestion
        const uut src_any{42};
```

Reply:
@pavel-kirienko is it fine if I fix this in separate pr? it will be just "cosmetic" pr without functional changes.

Reply:
Why can't we fix them now?

Reply:
yes we can - I just though it will be easier review separately "cosmetics" ONLY changes

Reply:
I don't think it makes a lot of difference here

Reply:
ok, I'll do it now, thnx

Reply:
done - see last commit 2ffc0051f5106b86dd7cdf5692f06c93fc311e35